### PR TITLE
[fix] Fixed apple silicon compile errors

### PR DESCRIPTION
**Pull request by poinwater**
# Description

Brief Description of the PR:
Fixed building errors for apple silicon devices

Fixes # (issue)
- Fixed broken code found during the apple silicon build test
- Updated the apple silicon build script
- Updated apple silicon's build instruction, unsupported APIs and frameworks on the README file

## Type of change

- [x] Bug fix
- [ ] New Tutorial
- [ ] Updated or additional documentation
- [ ] Additional Testing
- [ ] New Feature

# Checklist:

- [x] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/recommenders-addons/blob/master/CONTRIBUTING.md#coding-style)
    - [x] By running yapf
    - [x] By running clang-format
- [ ] This PR addresses an already submitted issue for TensorFlow Recommenders-Addons
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

# How Has This Been Tested?
Manually tested by calling test scripts on `dynamic_embedding/python/kernel_tests/`

### CODE REVIEW

**Comment by rhdong** [comment]
Hi @poinwater , could you squash the 8 commits to 1 or 2 commits according to the goals of them? Thank you !

File: tensorflow_recommenders_addons/dynamic_embedding/core/kernels/redis_impl/json.cc [hunk 1]
```diff
@@ -402,8 +402,8 @@ json_value *json_parse_ex(json_settings *settings, const json_char *json,
             case json_object:
 
               if (state.first_pass)
-                (*(json_char *
-                   __attribute((__may_alias__)) *)&top->u.object.values) +=
+                (*(json_char *__attribute((
```
Comment (rhdong) [inline tensorflow_recommenders_addons/dynamic_embedding/core/kernels/redis_impl/json.cc:405]: I believe you should use a mismatch format method or tools.

Comment (poinwater) [reply tensorflow_recommenders_addons/dynamic_embedding/core/kernels/redis_impl/json.cc:405]: I have used clang-format 13.0, I reformatted the code using clang-format 9.

**Comment by poinwater** [comment]
Hi @rhdong, 
I've squashed commits. I thought GitHub would squash the commits for me when merging to master because GitLab has the feature. But GitHub may not. lol

**Comment by rhdong** [comment]
> Hi @rhdong, I've squashed commits. I thought GitHub would squash the commits for me when merging to master because GitLab has the feature. But GitHub may not. lol

No matter, thank you!

**Pull request merged**